### ISSUE TITLE
[BUG] Errors encountered during the build process with the Clang toolchain

### ISSUE THREAD

**Issue by hahahahbenny** [gammasoft71/xtd#268]
# Describe the bug
I face two problems when building **xtd** with clang toolchain.
- **error: unknown type name 'nullptr_t'; did you mean 'std::nullptr_t'?**
- **error: no matching function for call to 'find'**

# To Reproduce

build command
> cmake --log-level=VERBOSE -DXTD_NATIVE_GRAPHIC_TOOLKIT=wxwidgets -DXTD_INSTALL_RESOURCES=OFF -DXTD_BUILD_TOOL_SLEEPFOR_COMMAND_LINE=OFF -DXTD_BUILD_SHARED_LIBRARIES=OFF -DXTD_INSTALL_EXAMPLES=OFF -DXTD_BUILD_TOOL_GUIDGEN_GUI=OFF -DXTD_BUILD_TOOL_GUIDGEN_COMMAND_LINE=OFF -DCMAKE_INSTALL_PREFIX=/home/bennywu/.xmake/packages/x/xtd/v0.1.2/245cf343678d4a779b164e08bd2649a0 -DCMAKE_INSTALL_LIBDIR:PATH=lib -G "Unix Makefiles" -DCMAKE_POSITION_INDEPENDENT_CODE=ON "-DCMAKE_CXX_FLAGS=-m64 -isystem /home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/include -isystem /usr/include/uuid" -DCMAKE_ASM_COMPILER=/usr/bin/clang -DCMAKE_AR=/usr/bin/ar "-DCMAKE_SHARED_LINKER_FLAGS=-m64 -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib/libcanberra-0.30 -lcanberra -lcanberra-null -lcanberra-alsa -lcanberra-oss -lcanberra-pulse -lcanberra-multi -luuid -ldl" "-DCMAKE_EXE_LINKER_FLAGS=-m64 -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib/libcanberra-0.30 -lcanberra -lcanberra-null -lcanberra-alsa -lcanberra-oss -lcanberra-pulse -lcanberra-multi -luuid -ldl" "-DCMAKE_CXX_LINK_EXECUTABLE=/usr/bin/clang++ <FLAGS> <CMAKE_CXX_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> -o <TARGET> <LINK_LIBRARIES>" -DCMAKE_STATIC_LINKER_FLAGS= -DCMAKE_RANLIB=/usr/bin/ranlib -DCMAKE_ASM_FLAGS=-m64 "-DCMAKE_C_FLAGS=-m64 -isystem /home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/include -isystem /usr/include/uuid" -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DCMAKE_C_COMPILER=/usr/bin/clang "-DCMAKE_C_FLAGS_RELEASE=-m64 -isystem /home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/include -isystem /usr/include/uuid -O3 -DNDEBUG" "-DCMAKE_CXX_FLAGS_RELEASE=-m64 -isystem /home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/include -isystem /usr/include/uuid -O3 -DNDEBUG" -DCMAKE_STATIC_LINKER_FLAGS_RELEASE= "-DCMAKE_EXE_LINKER_FLAGS_RELEASE=-m64 -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib/libcanberra-0.30 -lcanberra -lcanberra-null -lcanberra-alsa -lcanberra-oss -lcanberra-pulse -lcanberra-multi -luuid -ldl" "-DCMAKE_SHARED_LINKER_FLAGS_RELEASE=-m64 -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib -L/home/bennywu/.xmake/packages/l/libcanberra/0.30/6ccb04b6739b487b84dd680d74a73309/lib/libcanberra-0.30 -lcanberra -lcanberra-null -lcanberra-alsa -lcanberra-oss -lcanberra-pulse -lcanberra-multi -luuid -ldl"

# Expected behaviour

- fix the problems 

# Screenshots

![image](https://github.com/user-attachments/assets/53e6bdfe-c830-4d21-8b99-3643881c4374)

![image](https://github.com/user-attachments/assets/ef99af49-e02e-4993-9f21-aede7f21e0ff)

# Desktop (please complete the following information)

 - OS: ubuntu 
 - Version: 20.04 LTS
- xtd version: 0.1.2

**Additional context**

I have some queries regarding the code mentioned earlier.
- I wonder why there's an inconsistency in the use of nullptr_t within the code.
![image](https://github.com/user-attachments/assets/6c66dc31-8cfc-4a3a-9cf3-f4a3d520b10f)

**Comment by gammasoft71** [comment]
Hello,

First of all, thank you for your interest in xtd.

I will correct the problem as soon as possible.

The inconsistency is due to the fact that at one point I declared the following instruction `using namespace std;` in some `.cpp` files.
Then I removed the declaration of the std namespace, and unfortunately the compilers on some versions of C++ allow(allowed) the use of `nullptr_t` without specifying the `std::` namespace. 
Having had no compiler errors when removing the `using namespace std;` declaration, this remained unchanged in version 0.1.2. But this has been corrected in the trunk (I think).
Anyway I'll have a look and fix the problem.

**Comment by hahahahbenny** [comment]
Thanks for the quick reply！！You make sense. When I tested in the g++9 environment, I did not find these issues, but clang discovered that.





---Original---
From: ***@***.***&gt;
Date: Tue, Oct 29, 2024 20:45 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [gammasoft71/xtd] [BUG] Errors encountered during the buildprocess with the Clang toolchain (Issue #268)




 
Hello,
 
First of all, thank you for your interest in xtd.
 
I will correct the problem as soon as possible.
 
The inconsistency is due to the fact that at one point I declared the following instruction using namespace std; in some .cpp files.
 Then I removed the declaration of the std namespace, and unfortunately the compilers on some versions of C++ allow(allowed) the use of nullptr_t without specifying the std:: namespace.
 Having had no compiler errors when removing the using namespace std; declaration, this remained unchanged in version 0.1.2. But this has been corrected in the trunk (I think).
 Anyway I'll have a look and fix the problem.

**Comment by gammasoft71** [comment]
I have corrected the v0.1.0-beta branch with commit b34950b.
Can you check if it works for you and get back to me.
If it works, I'll create a new version 1.0.3.
To do this test you need to clone the v0.1.0 branch like this: 
```
git clone https://github.com/gammasoft71/xtd -b v0.1.0-beta xtd
```
Thanks for your help.
